shuup/gdpr/admin_module/views.py

```diff
@@ -0,0 +1,129 @@
+        ])
+        return toobar
+
+    def get_queryset(self):
+        return GDPRSettings.objects.filter(shop=get_shop(self.request))
+
+    def get_object(self):
+        return GDPRSettings.get_for_shop(get_shop(self.request))
+
+    def get_context_data(self, **kwargs):
+        context = super(GDPRView, self).get_context_data(**kwargs)
+        context["title"] = _("General Data Protection Regulation")
+        context["toolbar"] = self.get_toolbar()
+        return context
+
+    def form_valid(self, form):
+        result = self.save_form_parts(form)
+
+        gdpr_settings = self.get_object()
+        if gdpr_settings.enabled:
+            create_initial_privacy_policy_page(self.object.shop)
+            if not GDPRCookieCategory.objects.filter(shop=gdpr_settings.shop).exists():
+                create_initial_required_cookie_category(self.object.shop)
+        return result
+
+
+class BaseContactView(SingleObjectMixin, View):
+    def get_queryset(self):
+        queryset = Contact.objects.all()
+
+        limited = (settings.SHUUP_ENABLE_MULTIPLE_SHOPS and settings.SHUUP_MANAGE_CONTACTS_PER_SHOP and
+                   not self.request.user.is_superuser)
+        if limited:
+            queryset = queryset.filter(shops=get_shop(self.request))
+
+        return queryset
+
+
+class GDPRDownloadDataView(BaseContactView):
+    def post(self, request, *args, **kwargs):
+        contact = self.get_object()
+        contact.add_log_entry("User personal data download requested", kind=LogEntryKind.NOTE, user=self.request.user)
+        from shuup.gdpr.utils import get_all_contact_data
+        data = json.dumps(get_all_contact_data(contact))
+        response = HttpResponse(data, content_type="application/json")
+        filename = "attachment; filename=user_data_{}.json".format(now().strftime("%Y-%m-%d_%H:%M:%S"))
+        response["Content-Disposition"] = filename
+        return response
+
+
+class GDPRAnonymizeView(BaseContactView):
+    def post(self, request, *args, **kwargs):
+        contact = self.get_object()
+        contact.add_log_entry("User anonymization requested", kind=LogEntryKind.NOTE, user=self.request.user)
+        with atomic():
+            anonymizer = Anonymizer()
+
+            if isinstance(contact, PersonContact):
+                anonymizer.anonymize_person(contact)
+            elif isinstance(contact, CompanyContact):
+                anonymizer.anonymize_company(contact)
+
+            # check if there is any company related to the contact
+            # if so, anonymize it if he is the unique member
+            for company in contact.company_memberships.all():
+                if company.members.count() == 1:
+                    anonymizer.anonymize_company(company)
+                    # cancel company orders
+                    for order in Order.objects.incomplete().filter(customer=company):
+                        order.set_canceled()
+
+            if hasattr(contact, "user") and contact.user:
```

Comment:
Yeah really think this if should be inside the anonymize person. Is this ever called without that. And now it seems we have anonymization logic in multiple places which is not optimal for code readability.

Reply:
ok, I will unity them in a single place